src/extension.ts

```diff
@@ -118,14 +118,23 @@ function showSourceryStatusBarItem(context: ExtensionContext) {
   myStatusBarItem.show();
 }
 
-function registerNotifications(
-  languageClient: LanguageClient,
-  scanResultTree: ScanResultProvider,
-  scanResultTreeView: TreeView<TreeItem>,
-  chatProvider: ChatProvider,
-  recipeProvider: RecipeProvider,
-  context: ExtensionContext
-) {
+function registerNotifications({
+  languageClient,
+  scanResultTree,
+  scanResultTreeView,
+  chatProvider,
+  recipeProvider,
+  context,
+  troubleshootingProvider,
+}: {
+  languageClient: LanguageClient;
+  scanResultTree: ScanResultProvider;
+  scanResultTreeView: TreeView<TreeItem>;
+  chatProvider: ChatProvider;
+  recipeProvider: RecipeProvider;
+  context: ExtensionContext;
+  troubleshootingProvider: TroubleshootingProvider;
+}) {
```

Comment:
what does this change do?

Reply:
It just makes it so you can pass the arguments to the funciton in any order, a bit like keyword-ising it in python I guess.